index.js

```diff
@@ -182,11 +184,15 @@ function updatePanel(opts) {
   panel.show();
 }
 
+// Pass in a video URL as opts.src or pass in a video URL lookup function as opts.getUrlFn
 function launchVideo(opts) {
-  // opts {url: url, getUrlFn: getYouTubeUrl, domain: 'youtube.com'}
+  // opts {url: url,
+  //       getUrlFn: getYouTubeUrl,
+  //       domain: 'youtube.com',
+  //       src: streamURL or ''}
   const id = getVideoId(opts.url);
-  updatePanel({domain: opts.domain, id: id, src: ''});
-  opts.getUrlFn(id, function(err, streamUrl) {
+  updatePanel({domain: opts.domain, id: id, src: opts.src || ''});
+  opts.src ? updatePanel({src: opts.src}) : opts.getUrlFn(id, function(err, streamUrl) {
```

Comment:
This can be rewritten as:
```
if (!opts.src) {
  opts.getUrlFn(id, function(err, streamUrl) {
    if (!err) updatePanel({src: streamUrl});
  });
}
```

since on the line above calls `updatePanel` with `opts.src` we don't need to do it here.